bdd-generator/src/main/java/com/epam/test_generator/dao/interfaces/StepSuggestionDAO.java

```diff
@@ -15,7 +15,8 @@
 
     Page<StepSuggestion> findByProject(Project project, Pageable pageable);
 
-    Page<StepSuggestion> findByProjectIdAndContentIgnoreCaseContaining(Long projectId,
-                                                                       String searchString,
-                                                                       Pageable pageable);
+    Page<StepSuggestion> findByProjectIdAndContentIgnoreCaseContainingOrderByLastUsedDateDesc(
```

Comment:
This method is too long. You can reduce the length of this method by using appropriate names of parameters. Instead of saying "How does it work" you should name it by answering the question: "What's the business case?".

Reply:
I agree that name of this method is quite complicated and doesn't answer on business case. But, unfortunately, I can't change name of this method, since I haven't implemented it and Spring Data Jpa generates HQL query based on the name of method. Otherwise, I have to write SQL query.